### DISCUSSION TITLE
liquidation query

### DISCUSSION THREAD

**Discussion by deep0072** [Cyfrin/foundry-full-course-cu#670] [General]
lets  say alex is bad user has $140 of eth and $100 of minted.
so its undercollatralised situation.

now sam as a liquidator will pay burn $100 in protocol and get the reward is $140 worth of eth and 10 percent of bonus of burned coin in eth. right. if it is right.

then why we are 

```
     AggregatorV3Interface priceFeed = AggregatorV3Interface(s_collateralDeposited[tokenAddress]);

        (,,int256 price,) = priceFeed.latestRoundData();

        // $2000 is in 1eth
        // $500 eth? ==> 500/2000==> 0.25; 

        return (amountOfUSdInWei * PRECISION_FEED)/(price*ADDITIONAL_PRECISION_FEED);

    }




function liquidate(address tokenCollateralAddress, address User, uint256 debtToCOver) external moreThanZero(debtToCOver) nonReentrant {
        // need to check health factor first 

        if (_checkHealthFactor(user) >= 1){
            revert DscEngine__HealthFactorIsOk();
        }

        // as a liquidator we wanted to burn their dsc by depositin our eth equivalent of burn amount
        // then take collateral 
        // let say bad user = 140$ eth  and $100 dsc
        // deb to cover is $100 dsc that need to pay in eth as a liquidator
        // will get 140$ of eth or collateral

        uint256 tokenAmountFromDebtToRecover = getTokenfromUsd(tokenCollateralAddress, debtToCOver);

        // now liquidator will also get the bonus lets say bonus will be 10 percent of tokenAmountFromDebtToRecover

        uint256 bonusCollateral = (tokenAmountFromDebtToRecover * 10)/PRECISION;

        uint256 totalCollateral = tokenAmountFromDebtToRecover +  bonusCollateral;



    }```
    
    then why collateral amount is equivalent to  tokenAmountFromDebtToRecover  + bonusCollareal
                                                 ($100 + 10) = 110 why it is not 140 + 10 = 150

**Comment by 0xusmanf** [comment]
> now sam as a liquidator will pay burn $100 in protocol and get the reward is $140 worth of eth and 10 percent of bonus of burned coin in eth

Sam burned $100 worth of tokens so he will get back $100 worth of collateral + 10% of it ($100 + 10% = $110). He can not take on the total collateral of the bad user, the remaining collateral of the bad user will still be $30 after getting liquidated and no remaining debt = not a bad user anymore, He can redeem his $30 back.

**Reply by deep0072** [reply]
ok

**Reply by deep0072** [reply]
means his collateral got hit.that's what his (alex) disadvantage

**Comment by theirrationalone** [comment]
This might help you [discussion](https://github.com/Cyfrin/foundry-full-course-f23/discussions/523#discussioncomment-6630649) 

Also read this too: [comment](https://github.com/Cyfrin/foundry-full-course-f23/discussions/403#discussioncomment-6706197)

**Reply by deep0072** [reply]
thanks
